### PR TITLE
autorunner params from config

### PR DESCRIPTION
**Pull request by myron**
allows setting AutoRunner params from config
allows specifying number of folds in config

### CODE REVIEW

**Comment by wyli** [comment]
some issues with the integration tests
```
[2023-10-30T11:48:20.604Z] ======================================================================
[2023-10-30T11:48:20.604Z] ERROR: test_autorunner (__main__.TestAutoRunner)
[2023-10-30T11:48:20.604Z] ----------------------------------------------------------------------
[2023-10-30T11:48:20.604Z] Traceback (most recent call last):
[2023-10-30T11:48:20.604Z]   File "/home/jenkins/agent/workspace/Monai-notebooks/MONAI/tests/test_integration_autorunner.py", line 115, in test_autorunner
[2023-10-30T11:48:20.604Z]     runner = AutoRunner(
[2023-10-30T11:48:20.604Z]   File "/home/jenkins/agent/workspace/Monai-notebooks/MONAI/monai/apps/auto3dseg/auto_runner.py", line 238, in __init__
[2023-10-30T11:48:20.604Z]     self.export_cache()
[2023-10-30T11:48:20.604Z]   File "/home/jenkins/agent/workspace/Monai-notebooks/MONAI/monai/apps/auto3dseg/auto_runner.py", line 363, in export_cache
[2023-10-30T11:48:20.604Z]     ConfigParser.export_config_file(
[2023-10-30T11:48:20.604Z]   File "/home/jenkins/agent/workspace/Monai-notebooks/MONAI/monai/bundle/config_parser.py", line 444, in export_config_file
[2023-10-30T11:48:20.604Z]     with open(_filepath, "w") as f:
[2023-10-30T11:48:20.604Z] FileNotFoundError: [Errno 2] No such file or directory: '/tmp/tmpdt5w1bbf/work_dir/cache.yaml'
```

File: monai/apps/auto3dseg/auto_runner.py [hunk 1]
```diff
@@ -211,21 +211,11 @@ def __init__(
...
-        self.data_src_cfg = dict()
-        self.data_src_cfg_name = os.path.join(self.work_dir, "input.yaml")
-        self.algos = algos
-        self.templates_path_or_url = templates_path_or_url
-        self.allow_skip = allow_skip
-        self.kwargs = deepcopy(kwargs)
-
-        if input is None and os.path.isfile(self.data_src_cfg_name):
-            input = self.data_src_cfg_name
+        if input is None and os.path.isfile(os.path.join(os.path.abspath(work_dir), "input.yaml")):
```
Comment (mingxin-zheng) [inline monai/apps/auto3dseg/auto_runner.py:214]: I remember the `input.yaml` could be in other files specified by `input`

**Comment by myron** [comment]
> some issues with the integration tests
> 
> ```
> [2023-10-30T11:48:20.604Z] ======================================================================
> [2023-10-30T11:48:20.604Z] ERROR: test_autorunner (__main__.TestAutoRunner)
> [2023-10-30T11:48:20.604Z] ----------------------------------------------------------------------
> [2023-10-30T11:48:20.604Z] Traceback (most recent call last):
> [2023-10-30T11:48:20.604Z]   File "/home/jenkins/agent/workspace/Monai-notebooks/MONAI/tests/test_integration_autorunner.py", line 115, in test_autorunner
> [2023-10-30T11:48:20.604Z]     runner = AutoRunner(
> [2023-10-30T11:48:20.604Z]   File "/home/jenkins/agent/workspace/Monai-notebooks/MONAI/monai/apps/auto3dseg/auto_runner.py", line 238, in __init__
> [2023-10-30T11:48:20.604Z]     self.export_cache()
> [2023-10-30T11:48:20.604Z]   File "/home/jenkins/agent/workspace/Monai-notebooks/MONAI/monai/apps/auto3dseg/auto_runner.py", line 363, in export_cache
> [2023-10-30T11:48:20.604Z]     ConfigParser.export_config_file(
> [2023-10-30T11:48:20.604Z]   File "/home/jenkins/agent/workspace/Monai-notebooks/MONAI/monai/bundle/config_parser.py", line 444, in export_config_file
> [2023-10-30T11:48:20.604Z]     with open(_filepath, "w") as f:
> [2023-10-30T11:48:20.604Z] FileNotFoundError: [Errno 2] No such file or directory: '/tmp/tmpdt5w1bbf/work_dir/cache.yaml'
> ```

I updated to latest branch, and did a small change. But I'm not sure why it can't find cache.yaml.

**Comment by myron** [comment]
Ok, I believe I fixed it, and the integration test is working

**Review by wyli — APPROVED** [review]
Thanks, it looks good to me.

**Pull request merged**